tools/rust-tools/run-manager/src/main.rs

```diff
@@ -0,0 +1,437 @@
+            .get_account(&coordinator_instance)
+            .context("RPC error: failed to get account")?;
+
+        let instance = CoordinatorInstance::try_deserialize(&mut account.data.as_slice())
+            .context("Failed to deserialize CoordinatorInstance")?;
+
+        Ok(instance)
+    }
+
+    fn get_docker_tag_for_run(&self, local: bool, run_id: &str) -> Result<String> {
+        info!("Querying coordinator for Run ID: {}", run_id);
+
+        let instance = self.fetch_coordinator_data(run_id)?;
+
+        // Fetch the coordinator account to get the client version
+        let coordinator_account_data = self
+            .rpc_client
+            .get_account(&instance.coordinator_account)
+            .context("RPC error: failed to get coordinator account")?;
+
+        let coordinator_account = coordinator_account_from_bytes(&coordinator_account_data.data)
+            .context("Failed to deserialize CoordinatorAccount")?;
+
+        let client_version = String::from(&coordinator_account.state.client_version);
+
+        info!(
+            "Fetched CoordinatorInstance from chain: {{ run_id: {}, coordinator_account: {}, client_version: {} }}",
+            instance.run_id, instance.coordinator_account, client_version
+        );
+
+        let image_name = if local {
+            "psyche-solana-client".to_string()
+        } else {
+            "psyche-client".to_string()
+        };
+        let docker_tag = format!("{}:{}", image_name, client_version);
+        Ok(docker_tag)
+    }
+}
+
+/// Determine which Docker image to use and pull it if necessary
+async fn prepare_image(args: &Args, docker_mgr: &DockerManager) -> Result<String> {
+    let docker_tag = match &args.local {
+        // --local=version: use explicit version passed by parameter
+        Some(version) if !version.is_empty() => {
+            let tag = format!("psyche-solana-client:{}", version);
+            info!("Using explicit local version: {}", tag);
+            tag
+        }
+        // No --local or --local with no argument: query coordinator
+        _ => {
+            let run_id = get_env_var("RUN_ID")?;
+            let rpc = get_env_var("RPC")?;
+
+            let coordinator_program_id = args
+                .coordinator_program_id
+                .parse::<Pubkey>()
+                .context("Failed to parse coordinator program ID")?;
+            info!("Using coordinator program ID: {}", coordinator_program_id);
+
+            let coordinator = CoordinatorClient::new(rpc.clone(), coordinator_program_id);
+            let docker_tag = coordinator.get_docker_tag_for_run(args.local.is_some(), &run_id)?;
+            info!("Docker tag for run '{}': {}", run_id, docker_tag);
+            docker_tag
+        }
+    };
+
+    // Pull image unless --local flag is present
+    if args.local.is_none() {
+        docker_mgr.pull_image(&docker_tag)?;
+    } else {
+        info!("Using local image (skipping pull): {}", docker_tag);
+    }
```

Comment:
instead of checking this, you can directly return `Ok(docker_tag)` in the `let docker_tag = match  &args.local` branch above